### PR TITLE
Make __cdist_loglevel value more expressive.

### PR DESCRIPTION
**Pull request by darko-poljak**
With these changes you no longer have __cdist_loglevel values 40, 30, 10, etc. but "ERROR", "INFO", "DEBUG", etc.
`if [ "$__cdist_loglevel" = "DEBUG" ]`.

### CODE REVIEW

File: cdist/conf/type/__install_config/gencode-local [hunk 1]
```diff
@@ -24,16 +24,16 @@ remote_copy="$__type/files/remote/copy"
 
 cdist_args=""
 case "$__cdist_loglevel" in
```
Comment (asteven) [inline cdist/conf/type/__install_config/gencode-local]: If cdist itself honors an existing __cdist_loglevel than this is not required anymore.
Nested cdist invocations would automatically inherit this from the parents environment.

Comment (darko-poljak) [reply cdist/conf/type/__install_config/gencode-local]: But with current implementation only emulator honors this.
cdist itself, when started, does not.
I just replaced __debug with __cdist_loglevel.

**Comment by winduptoy** [comment]
@asteven @darko-poljak Sorry this is a little bit late after it's already been merged.

What if I wanted to log on all lower levels? For example, if I wanted to log on **INFO** and lower, I would be able to check `$__cdist_loglevel <= 20`. With a string I will only be able to check `$__cdist_loglevel = "INFO" || $__cdist_loglevel = "VERBOSE" || $__cdist_loglevel = "DEBUG" || $__cdist_loglevel = "TRACE"` which is much more difficult to reason about and do incorrectly than with integers.

**Comment by darko-poljak** [comment]
Hm... Then __cdist_log_level will have numerical value again, but we will
alao provide __cdist_log_level_name (INFO, DEBUG, TRACE, etc.).

** All work and no play makes Jack a dull boy **

Darko

**Pull request merged**